qiskit_experiments/curve_analysis/scatter_table.py

```diff
@@ -9,185 +9,434 @@
+        analysis: str | None = None,
+        check_unique: bool = True,
+    ) -> np.ndarray:
+        """Get subset of Y values.
+
+        A convenient shortcut of getting Y data with filtering.
+
+        Args:
+            kind: Identifier of the data, either data UID or name.
+            category: Name of data category.
+            analysis: Name of analysis.
+            check_unique: Set True to check if multiple series are contained.
+                When multiple series are contained, it raises a user warning.
+
+        Returns:
+            Numpy array of Y values.
+        """
+        sub_table = self.filter(kind, category, analysis)
+        if check_unique:
+            self._warn_composite_data(sub_table)
+        return sub_table.y
 
     @property
-    @deprecate_func(
-        since="0.6",
-        additional_msg="Curve data uses dataframe representation. Call .yerr.to_numpy() instead.",
-        pending=True,
-        package_name="qiskit-experiments",
-        is_property=True,
-    )
     def y_err(self) -> np.ndarray:
-        """Standard deviation of y values."""
-        return self.yerr.to_numpy()
+        """Standard deviation of Y values."""
+        return self._data.yerr.to_numpy(dtype=float, na_value=np.nan)
+
+    @y_err.setter
+    def y_err(self, new_values: np.ndarray):
+        self._data.loc[:, "yerr"] = new_values
+
+    def get_y_err(
+        self,
+        kind: int | str | None = None,
+        category: str | None = None,
+        analysis: str | None = None,
+        check_unique: bool = True,
+    ) -> np.ndarray:
+        """Get subset of standard deviation of Y values.
+
+        A convenient shortcut of getting Y error data with filtering.
+
+        Args:
+            kind: Identifier of the data, either data UID or name.
+            category: Name of data category.
+            analysis: Name of analysis.
+            check_unique: Set True to check if multiple series are contained.
+                When multiple series are contained, it raises a user warning.
+
+        Returns:
+            Numpy array of Y error values.
+        """
+        sub_table = self.filter(kind, category, analysis)
+        if check_unique:
+            self._warn_composite_data(sub_table)
+        return sub_table.y_err
 
     @property
-    @deprecate_func(
-        since="0.6",
-        additional_msg="Curve data uses dataframe representation. Call .shots.to_numpy() instead.",
-        pending=True,
-        package_name="qiskit-experiments",
-        is_property=True,
-    )
-    def shots(self):
-        """Shot number of data points."""
-        return self["shots"].to_numpy()
+    def name(self) -> np.ndarray:
+        """Corresponding data name."""
+        return self._data.name.to_numpy(dtype=object, na_value=None)
+
+    @name.setter
+    def name(self, new_values: np.ndarray):
+        self._data.loc[:, "name"] = new_values
+
+    @property
+    def class_id(self) -> np.ndarray:
```

Comment:
Any better name? `data_uid`?

Reply:
Done in [92cfc92](https://github.com/Qiskit-Extensions/qiskit-experiments/pull/1360/commits/92cfc92f5e0be9556155e25a1152a1c9abfdecf5)